sdk/core/azure-core/inc/credentials.hpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_CORE_IDENTITY_CREDENTIALS_HPP
+#define _az_CORE_IDENTITY_CREDENTIALS_HPP
+
+#include <chrono>
+#include <string>
+
+#include "_internal/prefix.hpp"
+
+namespace azure
+{
+
+namespace identity
+{
+
+class credential;
+
+namespace _internal
+{
+void set_scopes(credential& credential, std::string const& scopes);
+} // namespace _internal
+
+class credential
+{
+  friend void _internal::set_scopes(credential&, std::string const&);
+
+public:
+  virtual ~credential() noexcept {};
+
+protected:
+  credential() = default;
+
+private:
+  credential(credential const&) = delete;
+  void operator=(credential const&) = delete;
+
+  virtual void set_scopes(std::string const&) {}
+};
+
+ namespace _internal
+{
+
+ inline void set_scopes(credential& credential, std::string const& scopes)
+{
+  credential.set_scopes(scopes);
+}
+
+} // namespace _internal
+
+class token_credential : public credential
+{
+private:
+  std::string token_;
+  std::string scopes_;
+  std::chrono::steady_clock expires_at_;
+
+  token_credential(token_credential const&) = delete;
+  void operator=(token_credential const&) = delete;
+
+  virtual void set_scopes(std::string const& scopes) override { scopes_ = scopes; }
```

Comment:
`override` suggests this function is virtual in base class. So it's superfluous to use `virtual` in this case.

https://stackoverflow.com/questions/39932391/virtual-override-or-both-c

Reply:
Thank you! Done.